python/sqlalchemy/src/rest-service.py

```diff
@@ -16,10 +16,42 @@
 import traceback
 
 
+def init():
+
+    # try:
+    engine = create_engine('postgresql://{0}:{1}@{2}:{3}/postgres'.format(cfg.db_user, cfg.db_password, cfg.db_host, cfg.db_port))                            
+
+    existing_databases = engine.execute("SELECT datname FROM pg_database;")
+    # Results are a list of single item tuples, so unpack each tuple
+    existing_databases = [d[0] for d in existing_databases]
+
+    created_database = False
+    # Create database if not exists
+    if cfg.database not in existing_databases:
+        conn = engine.connect()
+        conn.execute('commit')
+        conn.execute("CREATE DATABASE {0};".format(cfg.database))
+        print("Created database {0}".format(cfg.database))
+        created_database = True
+
+    # Go ahead and use this engine
+    engine = create_engine('postgresql://{0}:{1}@{2}:{3}/{4}'.format(cfg.db_user, cfg.db_password, cfg.db_host, cfg.db_port, cfg.database))                            
+
+    if created_database == True:
+        conn = engine.connect()
+        conn.execute('commit')
+        conn.execute("CREATE SCHEMA {0}".format(cfg.schema))
+
+    Session = sessionmaker(ddwqbind=engine)
+    Base.metadata.create_all(engine)
+
+    # except Exception as e:
+    #     print('Here was the exception: %s' % e)
+
+
 logging.getLogger('sqlalchemy.engine.base.Engine').setLevel(logging.WARNING)
 
-engine = create_engine('postgresql://%s:%s@%s:%s/%s' % 
-                       (cfg.db_user, cfg.db_password, cfg.db_host, cfg.db_port, cfg.database), echo=True)
+engine = create_engine('postgresql://{0}:{1}@{2}:{3}/postgres'.format(cfg.db_user, cfg.db_password, cfg.db_host, cfg.db_port))                            
```

Comment:
we are not specifying the database here?

Reply:
We did not create the database yet at this point, so we connect to the default, then create the database, then create the schema (DB is created on line 35)